### ISSUE TITLE
PreserveDependencyAttribute is getting stripped when compiling individual assembly

### ISSUE THREAD

**Issue by stephentoub** [mono/linker#798]
The attribute and all uses of it is getting stripped when compiling corefx assemblies and applying the linker to each.  That means the dependency information doesn't survive to whole-app compilation, which means when linking the whole app, members reachable via reflection (where that dependency was specified via the attribute) can get removed erroneously.

cc: @marek-safar, @sbomer, @jkotas

**Comment by marek-safar** [comment]
This is by design, you need to pass `--keep-dep-attributes` to force keeping this attribute when linking

**Comment by stephentoub** [comment]
Thanks. Is that documented somewhere?

**Comment by marek-safar** [comment]
Yes, `monolinker --help` which I guess it's tricky for .net core
